### ISSUE TITLE
has_shipping_method( 'local_pickup' ) not working anymore because method_id is "local_pickup:3"

### ISSUE THREAD

**Issue by gerdneuman** [woothemes/woocommerce#11772]
After switching to the new shipping zones which came with 2.6 it some of my code is not working any more:
   
```
    public function is_local_pickup ( $order ) {
        // returns false suddenly
        // Debugging shows that the method_id is now "local_pickup:3"
        return $order->has_shipping_method( 'local_pickup' );
    }
```

I disabled all plugins and use current 2.6.4. 

By looking into my settings, I am guessing that 3 refers to the order (position) of the shipping method (free shipping is 1st, normal shipping is 2nd).

**Comment by gerdneuman** [comment]
@mikejolley BTW: This also caused another side effect for my theme (which I already fixed): On cart and checkout pages the `input#id` and `label[for]` was something like `shipping_method_local_pickup` IIRC but is now `shipping_method_0_local_pickup3`. I had some JS running on this which was broken after upgrading to the new shipping methods. Just telling since you've seemed interested in what causes problems during upgrades (as said, I already fixed this myself now). Changing html classes or IDs is such thing.

Anyway, thanks for the very fast fix.Looking forward to the release.
